epochStart/metachain/systemSCs_test.go

```diff
@@ -956,21 +961,119 @@ func createFullArgumentsForSystemSCProcessing(enableEpochsConfig config.EnableEp
 func TestSystemSCProcessor_ProcessSystemSmartContractInitDelegationMgr(t *testing.T) {
 	t.Parallel()
 
-	args, _ := createFullArgumentsForSystemSCProcessing(config.EnableEpochs{
-		StakingV2EnableEpoch: 1000,
-	}, testscommon.CreateMemUnit())
-	s, _ := NewSystemSCProcessor(args)
-
-	validatorsInfo := state.NewShardValidatorsInfoMap()
-	err := s.ProcessSystemSmartContract(validatorsInfo, &block.Header{})
-	assert.Nil(t, err)
+	t.Run("flag not active - activation epoch is in the future", func(t *testing.T) {
+		args, _ := createFullArgumentsForSystemSCProcessing(config.EnableEpochs{
+			DelegationSmartContractEnableEpoch: 39,
+			StakingV2EnableEpoch:               1000,
+		}, testscommon.CreateMemUnit())
+		s, _ := NewSystemSCProcessor(args)
+		handler := s.enableEpochsHandler.(enableEpochHandlerWithEpochConfirm)
+		handler.EpochConfirmed(37, 0)
+
+		validatorsInfo := state.NewShardValidatorsInfoMap()
+		err := s.ProcessSystemSmartContract(validatorsInfo, &block.Header{})
+		assert.Nil(t, err)
+
+		acc, err := s.userAccountsDB.GetExistingAccount(vm.DelegationManagerSCAddress)
+		assert.NotNil(t, err)
+		assert.True(t, check.IfNil(acc))
+	})
+	t.Run("flag active in that specific epoch", func(t *testing.T) {
+		args, _ := createFullArgumentsForSystemSCProcessing(config.EnableEpochs{
+			DelegationSmartContractEnableEpoch: 37,
+			StakingV2EnableEpoch:               1000,
+		}, testscommon.CreateMemUnit())
+		s, _ := NewSystemSCProcessor(args)
+		handler := s.enableEpochsHandler.(enableEpochHandlerWithEpochConfirm)
+		handler.EpochConfirmed(37, 0)
+
+		validatorsInfo := state.NewShardValidatorsInfoMap()
+		err := s.ProcessSystemSmartContract(validatorsInfo, &block.Header{})
+		assert.Nil(t, err)
+
+		acc, err := s.userAccountsDB.GetExistingAccount(vm.DelegationManagerSCAddress)
+		assert.Nil(t, err)
+
+		userAcc, _ := acc.(state.UserAccountHandler)
+		assert.Equal(t, userAcc.GetOwnerAddress(), vm.DelegationManagerSCAddress)
+		assert.NotNil(t, userAcc.GetCodeMetadata())
+	})
+	t.Run("flag not active - activation epoch is in the past", func(t *testing.T) {
+		args, _ := createFullArgumentsForSystemSCProcessing(config.EnableEpochs{
+			DelegationSmartContractEnableEpoch: 35,
+			StakingV2EnableEpoch:               1000,
+		}, testscommon.CreateMemUnit())
+		s, _ := NewSystemSCProcessor(args)
+		handler := s.enableEpochsHandler.(enableEpochHandlerWithEpochConfirm)
+		handler.EpochConfirmed(37, 0)
+
+		validatorsInfo := state.NewShardValidatorsInfoMap()
+		err := s.ProcessSystemSmartContract(validatorsInfo, &block.Header{})
+		assert.Nil(t, err)
+
+		acc, err := s.userAccountsDB.GetExistingAccount(vm.DelegationManagerSCAddress)
```

Comment:
if possible I would mock the systemVM and check specifically that the method is not called, instead of verifying that the account does not exist.

Reply:
in progress